### PR TITLE
fix: hide private user data in api/v3/users/[uid]

### PR DESCRIPTION
**Pull request by gasoved**

### CODE REVIEW

File: src/user/data.js [hunk 1]
```diff
@@ -141,6 +141,27 @@ module.exports = function (User) {
...
+		const privilegedOrSelf = isAdmin || isGlobalModerator || isSelf;
+
+		if (!privilegedOrSelf && (!userSettings.showemail || meta.config.hideEmail)) {
+			_userData.email = '';
+		}
+		if (!privilegedOrSelf && (!userSettings.showfullname || meta.config.hideFullname)) {
+			_userData.fullname = '';
+		}
+
+		return _userData;
```
Comment (julianlam) [inline src/user/data.js:162]: Missing removal of `ips` and `moderationNote` from this method (if this method is used in `getUserDataByUserSlug`, which I think it should

**Review by julianlam — CHANGES_REQUESTED** [review]
It would be good if the data removal logic from within `helpers.getUserDataByUserSlug` used the new `user.hidePrivateData` as well, so as to maintain DRY, what do you think?

**Comment by gasoved** [comment]
I thought it's a good enough place to not follow DRY. Use case is a bit different, and it would cause extra DB calls for both endpoints.

I'm pushing a commit doing that nevertheless. I'm in favor of the first version.
(This also adds two possible fields to /api/v3/users/:uid endpoint.)

**Comment by julianlam** [comment]
Merged via 97c8569a798075c50e93e585ac741ab55cb7c28b

**Pull request closed without merging**